scripts/build_climate_projections.py

```diff
@@ -0,0 +1,480 @@
+    """
+    Interpolate CMIP6 dataset to the cutout grid.
+
+    Parameters
+    ----------
+    cmip6_xr: xarray
+        CMIP6 climate projections loaded as xarray
+    cutout_xr : xarray of a cutout dataset
+        Cutout dataset loaded as xarray
+    """
+
+    # TODO read from the cutout file instead of hardcoding
+    dx_new = 0.3
+
+    newlon = np.arange(
+        round(np.min(cutout_xr.coords["x"].values), 1),
+        round(np.max(cutout_xr.coords["x"].values) + dx_new, 1),
+        dx_new,
+    )
+    newlat = np.arange(
+        round(np.min(cutout_xr.coords["y"].values), 1),
+        round(np.max(cutout_xr.coords["y"].values) + dx_new, 1),
+        dx_new,
+    )
+    cmip6_interp = cmip6_xr.interp(time=cmip6_xr.time, lat=newlat, lon=newlon)
+
+    return cmip6_interp
+
+
+def prepare_cmip6(cmip6_fl, cutout_xr):
+    """
+    Prepare the CMIP6 for calculation projections for the area of interest.
+
+    Parameters
+    ----------
+    cmip6_xr: xarray
+        CMIP6 climate projections loaded as xarray
+    cutout_xr : xarray of a cutout dataset
+        Cutout dataset loaded as xarray
+    """
+    cmip6_xr = xr.open_dataset(cmip6_fl)
+    cmip6_cropped_xr = crop_cmip6(cmip6_xr, cutout_xr)
+    cmip6_interp = interpolate_cmip6_to_cutout_grid(
+        cmip6_xr=cmip6_cropped_xr, cutout_xr=cutout_xr
+    )
+    return cmip6_interp
+
+
+# TODO fix years_window
+def subset_by_time(cmip6_xr, month, year, years_window):
+    """
+    Filter CMIP6 dataset according to the requested year and the processed
+    month.
+
+    Parameters
+    ----------
+    cmip6_xr: xarray
+        CMIP6 climate projections loaded as xarray
+    month: integer
+        A month value to be considered further for the morphing procedure
+    year: integer
+        The first year of the requested period in the future
+    years_window: integer
+        A width of the considered time period
+
+    Example
+    -------
+    month=3, year=2050, years_window=30
+    filters CMIP6 cmip6_xr dataset to calculate projection for March at 2050-2079
+    """
+
+    # TODO add warning in case there are no enough years
+    cmip6_in_period = cmip6_xr.where(
+        (
+            (cmip6_xr["time.month"] == month)
+            & (cmip6_xr["time.year"] >= year - years_window)
+            & (cmip6_xr["time.year"] < year + years_window - 1)
+        ),
+        drop=True,
+    )
+    return cmip6_in_period
+
+
+# TODO add functionality to customise CMIP6 ensemble
+def calculate_proj_of_average(
+    cmip6_xr, month, year0, year1, years_window, cmip6_param_name="t"
+):
+    """
+    Calculate a change in a monthly average value for a climate parameter for
+    the requested year periods and the processed month (a single value for each
+    grid cell)
+
+    Parameters
+    ----------
+    cmip6_xr: xarray
+        CMIP6 climate projections loaded as xarray
+    month: float
+        A month value to be considered further for the morphing procedure
+    year0: integer
```

Comment:
are year 0 and 1 like the base year and the prediction year?
the name could be more explicit

Reply:
Agree, revised. Are they more clear now?